spec/presenters/hyrax/iiif_manifest_presenter_spec.rb

```diff
@@ -0,0 +1,213 @@
+# frozen_string_literal: true
+
+# rubocop:disable BracesAroundHashParameters maybe a rubocop bug re hash params?
+RSpec.describe Hyrax::IiifManifestPresenter do
+  subject(:presenter) { described_class.new(work) }
+  let(:work) { build(:monograph) }
+
+  describe 'manifest generation' do
+    let(:builder_service) { Hyrax::ManifestBuilderService.new }
+
+    it 'generates a IIIF presentation 2.0 manifest' do
+      expect(builder_service.manifest_for(presenter: presenter))
+        .to include('@context' => 'http://iiif.io/api/presentation/2/context.json')
+    end
+
+    context 'with file set and work members' do
+      let(:work) { create(:work_with_image_files) }
+
+      it 'generates a manifest with nested content' do
+        expect(builder_service.manifest_for(presenter: presenter)['sequences'].first['canvases'].count)
+          .to eq 2
+      end
+
+      context 'and an ability' do
+        let(:ability) { Ability.new(user) }
+        let(:user) { create(:user) }
+
+        before { presenter.ability = ability }
+
+        it 'excludes items the user cannot read' do
+          expect(builder_service.manifest_for(presenter: presenter))
+            .not_to have_key('sequences')
+        end
+
+        it 'includes items with read permissions' do
+          readable = FactoryBot.create(:file_set, :image, user: user)
+          work.members << readable
+          work.save
+
+          expect(builder_service.manifest_for(presenter: presenter)['sequences'].first['canvases'].count)
+            .to eq 1
```

Comment:
How do we get to this number 1? Is there a property we can get from some of the involved objects?